test/unit/migrate_config_proxy_to_http_proxy.rb

```diff
@@ -0,0 +1,104 @@
+require 'test_plugin_helper'
+require File.join(Pathname.new(File.dirname(__FILE__)).parent.parent, 'db', 'migrate', '20200728054557_refer_config_proxy_to_http_proxy.rb')
+
+module ForemanVirtWhoConfigure
+  class MigrateConfigProxyToHttpProxy < ActiveSupport::TestCase
+    def migration_has_been_run?(version)
+      ActiveRecord::SchemaMigration.where(version: version).exists?
+    end
+
+    let(:migration) { klass.new }
+
+    describe "up" do
+      let(:config) { FactoryBot.build(:virt_who_config) }
+
+      before do
+        if migration_has_been_run?('20200728054557')
+          migration.down
+        end
+      end
+
+      it "should create Http Proxy if already not exist" do
+        config.proxy = "test.example.com"
+        config.save!
+
+        assert_difference('HttpProxy.count', 1) do
+          migration.up
+        end
+      end
+
+
+      it "should refer to existed Http Proxy" do
+        http_proxy = FactoryBot.create(:http_proxy, name: "test", url: "http://test.com")
+        config.proxy = "http://test.com"
+        config.save!
+        config_id = config.id
+
+        assert_difference('HttpProxy.count', 0) do
+          migration.up
+        end
+        
+        assert_equal updated_config.http_proxy_id, http_proxy.id
+      end
+
+      it "should refer to existed IP address http proxy" do
+        http_proxy = FactoryBot.create(:http_proxy, name: "test", url: "http:212.212.10.23")
+        config.proxy = "212.212.10.23"
+        config.save!
+        config_id = config.id
+
+        assert_difference('HttpProxy.count', 0) do
+          migration.up
+        end
+        
+        assert_equal updated_config.http_proxy_id, http_proxy.id
+      end
+
+      it "should create new http proxy if user and password is present" do
```

Comment:
I think we need 1 more test case when we have a `config.proxy = http://admin:changeme@example.com` - does it create the right HttpProxy with the credentials set?